lib/createExplorer.js

```diff
@@ -51,7 +54,7 @@ module.exports = function (options) {
       });
   }
 
-  function searchDirectory(directory) {
+  function searchDirectory(directory, lastSearchDirectory) {
```

Comment:
Do we need this argument? Maybe there is some logic that can avoid it. For example, what if we said:

If there is no next directory, 
  and this is not `stopDir`, 
  and this is not the home directory, 
then check the home directory.

Would that have the same effect without requiring `mustEnsureUserHome` or `lastSearchDirectory`?